nilearn/plotting/matrix_plotting.py

```diff
@@ -37,6 +40,151 @@ def fit_axes(ax):
         ax.set_position(new_position)
 
 
+def _sanitize_inputs_plot_matrix(mat_shape, tri, labels, reorder, figure, axes):  # noqa
+    """Helper function for plot_matrix.
+
+    This function makes sure the inputs to plot_matrix are valid.
+    """
+    _sanitize_tri(tri)
+    labels = _sanitize_labels(mat_shape, labels)
+    reorder = _sanitize_reorder(reorder)
+    fig, axes, own_fig = _sanitize_figure_and_axes(figure, axes)
+    return labels, reorder, fig, axes, own_fig
+
+
+def _sanitize_figure_and_axes(figure, axes):
+    """Helper function for plot_matrix."""
+    if axes is not None and figure is not None:
+        raise ValueError(
+            "Parameters figure and axes cannot be specified "
+            "together. You gave 'figure=%s, axes=%s'" % (figure, axes)
+        )
+    if figure is not None:
+        if isinstance(figure, plt.Figure):
+            fig = figure
+        else:
+            fig = plt.figure(figsize=figure)
```

Comment:
A bit weird to code figsize in figure...

Reply:
I agree, but the docstring of `plot_matrix` says you can do it:

```
figure : :class:`matplotlib.figure.Figure`, figsize :obj:`tuple`,\
        or None, optional
        Sets the figure used. This argument can be either an existing
        figure, or a pair (width, height) that gives the size of a
        newly-created figure.
```

If you think we shouldn't allow this, we'd have to deprecate it.

Reply:
This is a bit weird, but let's keep that outside of this PR ---given the existing docstrings.